integration_tests/src/main/python/cmp_test.py

```diff
@@ -144,3 +144,12 @@ def test_in(data_gen):
     assert_gpu_and_cpu_are_equal_collect(
             lambda spark : unary_op_df(spark, data_gen).select(f.col('a').isin(scalars)))
 
+# The difference between a set in and a regular in is the number of arguments, with the default
+# config setting 10 as the threashold.  Anything over 10 is an InSet
+@pytest.mark.parametrize('data_gen', eq_gens, ids=idfn)
+def test_in_set(data_gen):
+    # nulls are not supported for in on the GPU yet
+    scalars = list(gen_scalars(data_gen, 11, force_no_nulls=True))
```

Comment:
should this pull from the default config (somehow, maybe this isn't possible) and add one to it?

Reply:
It feels like a lot of extra work for very little benefit, but if you really want it I can do it.

Reply:
It is also an internal config so it could change at any point in time.

Reply:
> It is also an internal config so it could change at any point in time.

That's the point of referencing the config value.  If Spark changes the default the test will still be doing what it is intending to do, otherwise it potentially does not.

Reply:
No you don't understand the name of the config is just as likely to change as the default value is

Reply:
I don't agree the name of the config is just as likely to change as the value.  If the config changes then we're likely to not get a value for it.  At that point we could choose among a number of things to do, e.g.: fail the test to indicate we need to update this test work properly on the new Spark version.

Is this really "a lot of extra work"?  Are Spark configs not accessible from the data generator?

Reply:
There is no public API for us to get this config and apply the default.  I can read the config normally through `spark.conf`, but it will not do any of the default value processing.  If you just want me to move one hard coded 11 to another location with a hard coded 10 and add 1 to it I can, but if we actually want to know what spark has as the default I have to find a way to get a SQLConf from within python and call `optimizerInSetConversionThreshold` on it. If the name of that function changes I get a python error saying that it is not defined, which I then would have to figure out how to catch properly and fall back, or just let the test fail.

Reply:
It looks like if I get the sql context and call getConf on it I might get what we want. I'll see if I can use it

Reply:
Are you sure it doesn't do the default processing?  I just fired up a pyspark session with no custom conf settings, and it seems to be working for me:
```
Using Python version 3.7.6 (default, Jun  1 2020 18:57:50)
SparkSession available as 'spark'.
>>> spark.conf.get("spark.sql.optimizer.inSetConversionThreshold")
'10'
>>> spark.conf.get("spark.sql.shuffle.partitions")
'200'
>>> spark.conf.get("spark.sql.optimizer.planChangeLog.level")
'TRACE'
>>> 
```